cloud/blockstore/libs/endpoints/endpoint_manager.cpp

```diff
@@ -748,7 +756,7 @@ NProto::TStartEndpointResponse TEndpointManager::StartEndpointImpl(
     auto it = Endpoints.find(socketPath);
     if (it != Endpoints.end()) {
         auto endpoint = it->second;
-        auto error = AlterEndpoint(std::move(ctx), *request, *endpoint.Request);
+        auto error = RestartEndpoint(std::move(ctx), *request, *endpoint.Request);
```

Comment:
instead of refactoring AlterEndpoint and checking for existence of socket in the end.
Let's first ensure that socket exists (restarting endpoint if needed) and then invoke AlterEndpoint ... then no need to refactor AlterEndpoint at all
the code will look like this
```c++
    auto it = Endpoints.find(socketPath);
    if (it != Endpoints.end()) {
        auto endpoint = it->second;
	auto error = EnsureEndpointSocketExists(...)
        if (HasError(error)) {
            return TErrorResponse(error);
        }
				
        error = AlterEndpoint(std::move(ctx), *request, *endpoint.Request);
        if (HasError(error)) {
            return TErrorResponse(error);
        }
```

Reply:
There is a comparison of requests inside AlterEndpoint and it's a problem, because if nbs gets StartEndpointRequest with the same socket path but with different other arguments then we should do nothing and just return error "endpoint has already been started with other args". 
That is why I decided to refactor AlterEndpoint, and first of all compare the requests and then do AlterEndpoint and RecreateSocket if needed.

Reply:
Yep but regardless of the results of alter endpoint, in common case socket path should not be deleted for started endpoint. And  so it should be safe to stop/start endpoint with old parameters if needed before applying the alter endpoint logic